test/product-import.spec.coffee

```diff
@@ -470,3 +470,137 @@ describe 'ProductImport', ->
       updatedProduct = @import._ensureDefaults(sampleProduct)
       expect(updatedProduct).toEqual expectedProduct
 
+  describe ':: Custom reference resolution', ->
+
+    it ' :: should detect reference type attribute', ->
+      sampleReferenceAttribute =
+        name: 'foobar'
+        value:
+          value: 'some value'
+          resolvePredicate: 'some predicate query'
+          endpoint: 'some endpoint'
```

Comment:
This is coming from the project specific transformers, not in the data from customers.
So at the transformer level we know with what predicate a reference needs to be resolved.
So to give flexibility and to make it generic for the transformer, we decided to put in the predicate instead of just the resolving field.

About the `_resolveHelper` I did not understand.